render.go

```diff
@@ -351,11 +311,26 @@ func (r *Render) BreakLine(buffer *Buffer, lexer *Lexer) {
 	r.previousCursor = 0
 }
 
+func (r *Render) getCursorEndPos(text string, startPos int) int {
```

Comment:
Awesome! This function makes everything much easier. Can we have a description and tests for it?